app/views/events/_form.html.erb

```diff
@@ -147,11 +147,11 @@
       <div>
         <%= f.label :location_id, label: 'Event Location' %>
       </div>
-      <%= f.collection_select(:location_id,
-                              Location.includes(:region).available,
-                              :id,
-                              :name_with_region,
-                              {prompt: true}) %>
+      <%= f.select(:location_id,
+                   locations_for_select,
+                   include_blank: 'Please select',
```

Comment:
Sure thing, I'll write a test and fix the onchange. If it's alright with you, I'll also move the onchange function into a separate javascript file, since it's kind of complex to be in the template here. Thanks for your review.